internal/resources/ekscluster/data_source.go

```diff
@@ -79,6 +82,52 @@ func dataSourceTMCEKSClusterRead(ctx context.Context, d *schema.ResourceData, m
 			return true, nil
 		}
 
+		if isWaitForKubeconfig(d) {
+			clusFullName := &clustermodel.VmwareTanzuManageV1alpha1ClusterFullName{
+				Name:                  resp.EksCluster.Spec.AgentName,
+				OrgID:                 clusterFn.OrgID,
+				ManagementClusterName: "eks",
+				ProvisionerName:       "eks",
+			}
+			clusterResp, err := config.TMCConnection.ClusterResourceService.ManageV1alpha1ClusterResourceServiceGet(clusFullName)
+			// nolint: wsl
+			if err != nil {
+				log.Printf("Unable to get Tanzu Mission Control cluster entry, name : %s, error :  %s", clusterFn.Name, err.Error())
+				return true, err
+			}
+
+			mgmtClusterHealthy, err := isManagemetClusterHealthy(clusterResp)
+			if err != nil {
```

Comment:
nit: since this `if` and its `else if` have the same body, why not merge it?

Also, maybe returning the from that method error is not necessary, just bool should be fine?

Reply:
@asutoshpalai fixed it